appium-tests/common/common.spec.js

```diff
@@ -228,11 +228,18 @@ describe('Contacts Android', function () {
     function checkSession(done) {
         if (failedToStart) {
             fail('Failed to start a session');
-            done();
         }
+        done();
```

Comment:
I think you're still missing the purpose of this check 😄 
The way you put it, it wouldn't work as intended because it wouldn't skip the spec if called from beforeEach, it basically would check the variable and then call `beforeEach`'s `done()`. Then the spec would try to execute anyways.
The intent of this check is to skip the spec if the session is failed to start. We could also intentionally call `fail()` here to mark the spec as failed.